### ISSUE TITLE
TypeError: cannot use a string pattern on a bytes-like object (Python 3.5)

### ISSUE THREAD

**Issue by psykzz** [seatgeek/fuzzywuzzy#106]
```python
Traceback (most recent call last):
  File "/Users/msmith/Desktop/tmp/slack_Stuff/venv/lib/python3.5/site-packages/fuzzywuzzy/process.py", line 102, in extract
    for key, choice in choices.items():
AttributeError: 'list' object has no attribute 'items'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "slackbot.py", line 170, in <module>
    bot.process()
  File "slackbot.py", line 154, in process
    self.handle_event(reply)
  File "slackbot.py", line 130, in handle_event
    plugin.call(event['func'], event)
  File "/Users/msmith/Desktop/tmp/slack_Stuff/plugins.py", line 85, in call
    func(event)
  File "/Users/msmith/Desktop/tmp/slack_Stuff/plugins.py", line 139, in on_message
    game = self.get_steam_game(game_name)
  File "/Users/msmith/Desktop/tmp/slack_Stuff/plugins.py", line 116, in get_steam_game
    matched_game, score = process.extract(game_name, [game['name'] for game in game_list], limit=1)[0]
  File "/Users/msmith/Desktop/tmp/slack_Stuff/venv/lib/python3.5/site-packages/fuzzywuzzy/process.py", line 109, in extract
    processed = processor(choice)
  File "/Users/msmith/Desktop/tmp/slack_Stuff/venv/lib/python3.5/site-packages/fuzzywuzzy/utils.py", line 84, in full_process
    string_out = StringProcessor.replace_non_letters_non_numbers_with_whitespace(s)
  File "/Users/msmith/Desktop/tmp/slack_Stuff/venv/lib/python3.5/site-packages/fuzzywuzzy/string_processing.py", line 25, in replace_non_letters_non_numbers_with_whitespace
    return cls.regex.sub(u" ", a_string)
TypeError: cannot use a string pattern on a bytes-like object
```

**Comment by josegonzalez** [comment]
Do you know what the input string was?

**Comment by psykzz** [comment]
I've reproduced it with a number of strings but lets assume its `eve online`

**Comment by josegonzalez** [comment]
Looks like [we expect unicode/string objects](https://github.com/seatgeek/fuzzywuzzy/blob/master/fuzzywuzzy/string_processing.py#L25), not raw bytes.

We could decode your byte string into unicode in the package. Not sure thats the best way to handle this input, and I'm almost of the opinion that we should only be handling unicode string objects.

@acslater00 any strong feelings on this? I think I may be out of my depth.

**Comment by psykzz** [comment]
You shouldn't need to define your empty string as `u" "` as you already include `
from __future__ import unicode_literals` at the top of the [file](https://github.com/seatgeek/fuzzywuzzy/blob/master/fuzzywuzzy/string_processing.py#L1)

**Comment by psykzz** [comment]
How about this PR https://github.com/seatgeek/fuzzywuzzy/pull/107

**Comment by josegonzalez** [comment]
Ooo fancy. I'll take a look when I get into the office, good catch!

**Comment by josegonzalez** [comment]
Good show. Just merged, will release in a bit.

**Comment by josegonzalez** [comment]
This was released in 0.9.0, though the release script dropped the commit message because *shrug*. Thanks for the pr though!
